src/standalone/Soil/rre.jl

```diff
@@ -179,6 +179,16 @@ function ClimaLand.make_compute_imp_tendency(model::RichardsModel)
         # GradC2F returns a Covariant3Vector, so no need to convert.
         @. dY.soil.ϑ_l =
             -(divf2c_water(-interpc2f(p.soil.K) * gradc2f_water(p.soil.ψ + z)))
+
+        # Source terms, also update the vertical integral of the source: p.soil.∫S_θ_liq_dz
+        p.soil.∫S_θ_liq_dz .= 0
+        # These change ∫S and dY by +=, which is why we zero them out above.
+        for src in model.sources
+            if !src.explicit
+                ClimaLand.source!(dY, src, Y, p, model)
+            end
+        end
+        @. dY.soil.∫F_vol_liq_water_dt += p.soil.∫S_θ_liq_dz # these source terms are stepped implicitly
```

Comment:
Why not update `dY` inside `source!`?

Reply:
(same for the explicit sources and for other models)

Reply:
Okay! I hadn't thought about it before, but it might be nice to have all source contributions to `dY` done inside of `source!` (any maybe for `source!` to not modify `p` if possible but not necessarily :) )

Reply:
yes! I think this works out. nice! testing now and then will push